eis_toolkit/raster_processing/unique_combinations.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+import rasterio
+from beartype import beartype
+from beartype.typing import List, Tuple
+
+from eis_toolkit.exceptions import InvalidParameterValueException
+from eis_toolkit.raster_processing.check_raster_grids import check_raster_grids
+
+
+def _unique_combinations(
+    bands: List[np.ndarray],
+) -> np.ndarray:
+
+    combined_array = np.stack(bands, axis=-1)
+
+    combined_array_2d = combined_array.reshape(-1, len(bands))
+
+    _, unique_indices, inverse_indices = np.unique(combined_array_2d, axis=0, return_index=True, return_inverse=True)
+
+    unique_combinations = inverse_indices.reshape(bands[0].shape)
+
+    unique_combinations += 1
+
+    return unique_combinations
+
+
+@beartype
+def unique_combinations(  # type: ignore[no-any-unimported]
+    raster_list: List[rasterio.io.DatasetReader],
```

Comment:
You could change the `List` annotation into a `Sequence`, just so that beartype won't get mad if a user will give a tuple. Also change this in the inner function